histoqc/config/__init__.py

```diff
@@ -34,5 +34,8 @@ def read_config_template(name=None):
     """return the contents of a configuration template"""
     templates = list_config_templates()
     if name not in templates:
-        raise KeyError(f'no configuration template found under key {name!r}')
```

Comment:
I think you're working too hard here, i suspect its going to be a lot easier and more robust to instead modify CONFIG_TEMPLATE_RE to accept all desired config files, instead of hacking around its failure at this stage

Reply:
The regular expressions obtain the template name, for example config_clinical.ini, template name is clinical.
For config_v2.1.ini , template name is v2.1
Will have change in command - python -m histoqc -c v2.1.ini -n 3 "*.svs"
Is that fine? The current fix lets it to be used as is

Reply:
Interesting! I hadn't noticed or considered that....

That said, it doesn't seem to work:
```
D:\temp\HistoQC>python -m histoqc -c v2.1.ini "D:\research\cetl_vs_mct\B15_6130_2_CETL.svs"
2021-09-23 11:29:44,558 - WARNING - Configuration file v2.1.ini not found, checking for a template match.
Traceback (most recent call last):
  File "D:\temp\HistoQC\histoqc\__main__.py", line 80, in main
    config.read_file(open(args.config))
FileNotFoundError: [Errno 2] No such file or directory: 'v2.1.ini'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "D:\temp\HistoQC\histoqc\__main__.py", line 229, in <module>
    sys.exit(main())
  File "C:\Python37\lib\contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "D:\temp\HistoQC\histoqc\__main__.py", line 83, in main
    config.read_string(read_config_template(args.config))
  File "D:\temp\HistoQC\histoqc\config\__init__.py", line 40, in read_config_template
    raise KeyError(f'no configuration template found under key {name!r}')
KeyError: "no configuration template found under key 'v2.1.ini'"

```

but I think based on what you said, the name should actually be "v2.1" and not "v.2.1.ini" (minus file extension)?

Regardless that also doesn't work unfortunately

```
D:\temp\HistoQC>python -m histoqc -c v2.1 "D:\research\cetl_vs_mct\B15_6130_2_CETL.svs"
2021-09-23 11:29:36,371 - WARNING - Configuration file v2.1 not found, checking for a template match.
Traceback (most recent call last):
  File "D:\temp\HistoQC\histoqc\__main__.py", line 80, in main
    config.read_file(open(args.config))
FileNotFoundError: [Errno 2] No such file or directory: 'v2.1'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "D:\temp\HistoQC\histoqc\__main__.py", line 229, in <module>
    sys.exit(main())
  File "C:\Python37\lib\contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "D:\temp\HistoQC\histoqc\__main__.py", line 83, in main
    config.read_string(read_config_template(args.config))
  File "D:\temp\HistoQC\histoqc\config\__init__.py", line 40, in read_config_template
    raise KeyError(f'no configuration template found under key {name!r}')
KeyError: "no configuration template found under key 'v2.1'"


```


But this is a highly related question which I think also needs to be addressed and doing so would address the issue above:
why doesn't the config_v2.1.ini appear here, if it is in the directory?

```
D:\research\HistoQC>python -m histoqc.config --list
# available example configurations:
- default:  config.ini
- clinical: config_clinical.ini
- first:    config_first.ini
- ihc:      config_ihc.ini
- light:    config_light.ini
```

Reply:
Currently the regular expression does not account for decimal points, and will have to add it in,
Once added it will pick the name v2.1 and it will appear as a template name.

Should I go ahead and change it ?

Reply:
yes, i think so, thanks

Reply:
now we are adding a decimal point as well in the CONFIG_TEMPLATE_RE
CONFIG_TEMPLATE_RE = re.compile(r'config(_(?P<name>[A-Za-z][A-Za-z0-9.]+))?[.]ini')

>python -m histoqc.config --list**
# available example configurations:
- default:  config.ini
- clinical: config_clinical.ini
- first:    config_first.ini
- ihc:      config_ihc.ini
- light:    config_light.ini
- v2.1:     config_v2.1.ini

Hence now the user can use the command 
> python -m histoqc -c v2.1 -n 3 "*.svs" and it will work fine.

Reply:
Love it!